awx/main/fields.py

```diff
@@ -790,13 +790,14 @@ def schema(self, model_instance):
                 'extra_vars': {
                     'type': 'object',
                     'patternProperties': {
-                        r'^(?:(?:{(?:{|%)[^{}]*?(?:%|})})|(?:[a-zA-Z_]+[a-zA-Z0-9_]*)+)+$': {
-                            "anyOf": [{'type': 'string'}, {'type': 'array'}, {'$ref': '#/properties/extra_vars'}]
-                        }
+                        # http://docs.ansible.com/ansible/playbooks_variables.html#what-makes-a-valid-variable-name
+                        # plus, add ability to template
+                        r'^[a-zA-Z_\{\}]+[a-zA-Z0-9_\{\}]*$': {"anyOf": [{'type': 'string'}, {'type': 'array'}, {'$ref': '#/properties/extra_vars'}]}
```

Comment:
I found that a lot of error cases got caught by validation _here_ when I wanted it to get caught by the nicer validation in `validate_extra_vars`. So this is trying to make the regex do less. There may still be templating of the keys, but keys must be strings.